transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -162,7 +162,7 @@ NativeDatagramPacketArray cleanDatagramPacketArray() {
 
     @Override
     protected void wakeup(boolean inEventLoop) {
-        if (!inEventLoop && WAKEN_UP_UPDATER.compareAndSet(this, 0, 1)) {
+        if (!inEventLoop && WAKEN_UP_UPDATER.getAndSet(this, 1) == 0) {
             // write to the evfd which will then wake-up epoll_wait(...)
             Native.eventFdWrite(eventFd.intValue(), 1L);
```

Comment:
Doesn't this essentially mean that every invocation of wakeup from outside the event loop will trigger eventFdWrite even if wakenUp is 1? Can we add this to the benchmark?

Reply:
@johnou no, this is functionally equivalent. If `wakenUp` is 1, then `getAndSet` will return 1 and so the if clause will still evaluate to false.

Reply:
oh man I missed the == 0, I need to stop doing reviews from the phone..